### PR TITLE
RATY-116 vulnerable dependencies

### PR DESCRIPTION
**Pull request by mikkojamG**
Fixing vulnerabilities and removing deprecated packages.
- Add nvmrc
- Replace deprecated oidc-client
- Upgrade HDS packages to latest version, replacing deprecated Navigation and Footer components
- Replace CRA + Jest with Vite and Vitest, this also fixes multiple vulnerabilities

[RATY-116](https://helsinkisolutionoffice.atlassian.net/browse/RATY-116)

[RATY-116]: https://helsinkisolutionoffice.atlassian.net/browse/RATY-116?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Review by arpargo — CHANGES_REQUESTED** [review]

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**PYSAKOINNIN-SAHK-ASIOINTI-UI branch is deployed to platta: https://pysakoinnin-sahk-asiointi-pr153-ui.dev.hel.ninja** :rocket::rocket::rocket:

File: .env.test
[the remark this answers is not in the document]
Comment (mikkojamG) [reply .env.test:1]: [154a929](https://github.com/City-of-Helsinki/pysakoinnin-sahk-asiointi-ui/pull/153/commits/154a929f3d18512b0b3bbc138723ac422443f3b1)

**Review by arpargo — APPROVED** [review]
LGTM

**Pull request merged**